### DISCUSSION TITLE
Studio v3 ignoring `basePath` on build.

### DISCUSSION THREAD

**Discussion by jozefini** [sanity-io/sanity#3592] [Sanity Studio Support] [answered]
Hello folks,

I'm having a hard time making the studio work as sub path `/studio` on Next.js. This is not new to me, I've done on all v2 studios but idk yet how to do with the v3.

When I run `yarn start` works great. The studio is shown on `/studio` however when I build the studio, copy it to `public` folder in Next.js. When I try to access `localhost:3000/studio` it shows white screen.

On console I see 404 errors trying to fetch assets directly on root like `http://localhost:3000/static/sanity.0b5333cf.js` without `/studio` basepath, the correct URL should be `http://localhost:3000/studio/static/sanity.0b5333cf.js` to work.

My config is like this:
```
export default createConfig({
  name: 'default',
  title: 'Dashboard',
  basePath: '/studio', // here is the base path.
  projectId: 'xxxxxxx',
  dataset: 'production',
  plugins: [
    deskTool(),
    visionTool({
      defaultApiVersion: 'v2021-10-21',
      defaultDataset: 'production',
    }),
  ],
  schema: {
    types: schemaTypes,
  },
})
```

**Comment by lobunto** [comment, marked as answer]
@jozefini I had the same issue, and I think you have to set the `basePath` in `sanity.cli.[js|ts]` as well:

```ts
import { createCliConfig } from 'sanity/cli';

export default createCliConfig({
  api: {
    projectId: process.env.SANITY_STUDIO_API_PROJECT_ID,
    dataset: process.env.SANITY_STUDIO_API_DATASET,
  },
  project: {
    basePath: '/studio',
  },
});
```

**Reply by jozefini** [reply]
That would make sense. I will give it a try.

Thank you for sharing!

**Comment by jh3y** [comment]
Unfortunately, I had no joy with this either running on `v3.0.0`. I have a `basePath` in both places. But, when I go to `build` on the CLI, the output points to `/static` instead of `/studio/static`.

**Reply by jh3y** [reply]
I see there's another issue open for this: https://github.com/sanity-io/sanity/issues/3972 👍
